segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -496,12 +497,23 @@ public void initialize() throws Exception {
      */
     private CompletableFuture<Void> writeSnapshotInfoIfRequired() {
         // Save if we have generated newer snapshot since last time we saved.
-        if (lastSavedSystemSnapshot.get() != null && lastSavedSnapshotInfo.get() != null
-                && lastSavedSnapshotInfo.get().getSnapshotId() < lastSavedSystemSnapshotId.get()) {
-            return writeSnapshotInfo(lastSavedSystemSnapshotId.get());
-        } else {
-            return CompletableFuture.completedFuture(null);
+        if (lastSavedSystemSnapshot.get() != null) {
+            boolean shouldSave = true;
+            if (lastSavedSnapshotInfo.get() == null) {
+                log.debug("SystemJournal[{}] Saving first snapshot info new={}.", containerId, lastSavedSystemSnapshotId.get());
+            } else if (lastSavedSnapshotInfo.get().getSnapshotId() < lastSavedSystemSnapshotId.get()) {
+                log.debug("SystemJournal[{}] Saving new snapshot info new={} old={}.", containerId,
+                        lastSavedSystemSnapshotId.get(), lastSavedSnapshotInfo.get().getSnapshotId());
+            } else {
+                shouldSave = false;
+                log.trace("SystemJournal[{}] Snapshot info is up to date", containerId);
```

Comment:
We rarely enable trace. Would it be ok to make this debug?

Reply:
It will create lot of useless spam.
I think I'll just remove it.